contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2319,16 +2319,11 @@ object_id(PG_FUNCTION_ARGS)
 			else if (!strcmp(object_type, "af") || !strcmp(object_type, "fn") || !strcmp(object_type, "fs") ||
 					 !strcmp(object_type, "ft") || !strcmp(object_type, "if") || !strcmp(object_type, "p") ||
 					 !strcmp(object_type, "pc") || !strcmp(object_type, "tf") || !strcmp(object_type, "rf") ||
-					 !strcmp(object_type, "x"))
+					 !strcmp(object_type, "x") || !strcmp(object_type, "tr") || !strcmp(object_type, "ta"))
 			{
 				/* search in pg_proc by name and schema oid */
 				result = tsql_get_proc_oid(object_name, schema_oid, user_id);
 			}
-			else if (!strcmp(object_type, "tr") || !strcmp(object_type, "ta"))
-			{
-				/* search in pg_trigger by name and schema oid */
-				result = tsql_get_trigger_oid(object_name, schema_oid, user_id);
-			}
```

Comment:
For object_id('procedure_name', 'tr') we should & object_id('trigger_name', 'p') we should return NULL.